knowledge_graph/classifier/bert_based.py

```diff
@@ -148,71 +179,206 @@ def _prepare_dataset(self, labelled_passages: list[LabelledPassage]) -> Dataset:
         Returns:
             Dataset: A HuggingFace dataset ready for training
         """
-        texts = []
-        labels = []
-
-        for passage in labelled_passages:
-            contains_concept = any(
+        texts = [passage.text for passage in labelled_passages]
+        labels = [
+            1
+            if any(
                 span.concept_id == self.concept.wikibase_id for span in passage.spans
             )
-            texts.append(passage.text)
-            labels.append(1 if contains_concept else 0)
+            else 0
+            for passage in labelled_passages
+        ]
 
         tokenized_inputs = self.tokenizer(  # type: ignore[operator]
-            texts, padding="max_length", truncation=True, return_tensors="pt"
+            texts, padding=True, truncation=True, max_length=512, return_tensors=None
```

Comment:
_nitpick:_ it'd be good to document where this 512 comes from (to match with the beginner-friendly documentation elsewhere), and whether it's related to the choice of pretrained model